client/resource_group/controller/controller.go

```diff
@@ -1044,6 +1055,28 @@ func (gc *groupCostController) collectRequestAndConsumption(selectTyp selectType
 	return req
 }
 
+// If the resource group has background jobs, we should not record consumption and wait for it.
+func (gc *groupCostController) isBackgroundRequest(requestResource string) bool {
+	gc.metaLock.Lock()
+	defer gc.metaLock.Unlock()
+
+	if gc.meta.BackgroundSettings != nil {
+		jobTypes := gc.meta.BackgroundSettings.JobTypes
+		jobTypesMap := make(map[string]struct{})
+		for _, jobType := range jobTypes {
+			jobTypesMap[jobType] = struct{}{}
+		}
+
+		parts := strings.Split(requestResource, "_")
+		for _, part := range parts {
+			if _, ok := jobTypesMap[part]; ok {
+				return true
+			}
+		}
```

Comment:
```suggestion
		if len(requestResource) == 0 || len(gc.meta.BackgroundSettings.JobTypes) == 0 {
				return false;
		}
		parts := strings.Split(requestResource, "_")
		return slices.Contains(gc.meta.BackgroundSettings.JobTypes, parts[len(parts)-1])
```

Reply:
Check for all parts due to this https://github.com/pingcap/tidb/pull/44950/files#diff-b1e739aabb5ea3cbc8cac2e5df62c017e733983caf39b090e094f48091c9eadbR263

Reply:
you should depend on nothing but the explict_source_type in the request source. Any component uses the background resource control should explicitly set the right session variable or default value.

Reply:
Okay, I will fix tidb also.